src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+                    break;
+                case 20:
+                    this->unk_1EC = 0x12;
+                    break;
+                case 21:
+                    this->unk_1EC = 0x14;
+                    break;
+                case 22:
+                    this->unk_1EC = 0x15;
+                    break;
+                case 23:
+                    this->unk_1EC = 0x17;
+                    break;
+                case 24:
+                    this->unk_1EC = 0x18;
+                    break;
+                default:
+                    this->unk_1EC = 0;
+                    break;
+            }
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+        }
+
+        if ((this->unk_1EC == 5) && (globalCtx->sceneNum == 8) && (gSaveContext.sceneSetupIndex == 0xB) &&
+            (globalCtx->csCtx.frames == 0x190)) {
+            Actor_PlaySfxAtPos(&this->actor, 0x697D);
+        }
+        if (this->unk_1EC == 0x12) {
+            func_80AD0998(this);
+        }
+        if (this->unk_1EC == 0x15) {
+            func_80AD0A24(this);
+        }
+        if (this->unk_1EC == 0x18) {
+            func_80AD0AB0(this);
+        }
+        if ((this->unk_1EC == 0x14) &&
+            (((Animation_OnFrame(&this->skelAnime, 17.0f))) || (Animation_OnFrame(&this->skelAnime, 27.0f)) ||
+             (Animation_OnFrame(&this->skelAnime, 37.0f)) || (Animation_OnFrame(&this->skelAnime, 47.0f)) ||
+             (Animation_OnFrame(&this->skelAnime, 57.0f)) || (Animation_OnFrame(&this->skelAnime, 67.0f)))) {
+            Actor_PlaySfxAtPos(&this->actor, 0x29B3);
+        }
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+        return;
+    }
+    this->unk_1F0 = 1;
+    this->unk_1ED = 0x63;
+    func_80AD144C(this, globalCtx);
+}
+
+void func_80AD19A0(EnOsn* this, GlobalContext* globalCtx) {
+    u8 temp_v0 = Message_GetState(&globalCtx->msgCtx);
+
+    if ((temp_v0 == 6 || temp_v0 == 5) && func_80147624(globalCtx)) {
+        if (this->unk_1EA & 0x20) {
+            this->unk_1EA &= 0xFFDF;
+            globalCtx->msgCtx.unk11F22 = 0x43;
+            globalCtx->msgCtx.unk12023 = 4;
+            this->actionFunc = func_80AD14C8;
+        } else {
+            func_80AD10FC(this, globalCtx);
+        }
+    }
+}
+
+void EnOsn_Idle(EnOsn* this, GlobalContext* globalCtx) {
+}
+
+void EnOsn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnOsn* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_osn_Skel_0202F0, &object_osn_Anim_0201BC, 0, 0, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->unk_1FC = -1;
```

Comment:
```suggestion
    this->unk_1FC = 255;
```
`unk_1FC` is unsigned and an alpha value, so the number should not be negative